src/OSPSuite.UI/Views/Charts/CurveSettingsView.cs

```diff
@@ -378,5 +382,41 @@ private void gridProcessGridKey(KeyEventArgs e)
 
          _presenter.Remove(curve);
       }
+
+      private void onPopupMenuShowing(object sender, PopupMenuShowingEventArgs e)
+      {
+         var gridViewMenu = e.Menu;
+         if (gridViewMenu == null)
+            return;
+
+         //only one or no curve selected. We do not show the option
+         if (gridView.GetSelectedRows().Length <= 1)
+            return;
+
+         var copyAllMenu = new DXMenuItem(Captions.EditAllCurvesProperties, (o, args) => onEditProperties(), ApplicationIcons.Edit);
+         gridViewMenu.Items.Insert(0, copyAllMenu);
+      }
+
+      private void onEditProperties()
+      {
+         var selectedCurveList = getSelectedCurveDTOs();
+
+         if (selectedCurveList.Count > 1)
+            _presenter.EditProperties(selectedCurveList);
+      }
+
+      private List<CurveDTO> getSelectedCurveDTOs()
+      {
+         var selectedCurveList = new List<CurveDTO>();
+         for (var i = 0; i < gridView.SelectedRowsCount; i++)
+         {
+            var row = (gridView.GetSelectedRows()[i]);
```

Comment:
I don't like this code. you are calling GetSelect4edRows on each iteration. Cache it before iterating.
Probably there is a way to use a foreach here.

Reply:
done and also converted to foreach